src/rpc/daemon_handler.cpp

```diff
@@ -921,8 +922,11 @@ namespace rpc
       if (matched_handler == std::end(handlers) || matched_handler->method_name != request_type)
         return BAD_REQUEST(request_type, req_full.getID());
 
-      std::string response = matched_handler->call(*this, req_full.getID(), req_full.getMessage());
-      MDEBUG("Returning RPC response: " << response);
+      epee::byte_slice response = matched_handler->call(*this, req_full.getID(), req_full.getMessage());
+
+      const boost::string_ref response_view{reinterpret_cast<const char*>(response.data()), response.size()};
```

Comment:
Could be moved inside `MDEBUG(...)`, so it doesn't get evaluated of debug logs are disabled.

Reply:
That's a huge line to save a pointer copy and length copy. Since all of those definitions are visible in the header, the compiler could elide/delay through analysis already. I'll see how I clean this up though.

Reply:
I made this change with a multi-line debug statement. Maybe more eye-pleasing too.

Reply:
Ok, now I remember why I did it this way. The comment for the `boost::string_ref` constructor breaks the `MDEBUG` macro which interprets it as an additional macro argument. It should be possible to "fix" the macro, but this temporary object is fairly light-weight to construct.

And it just occurred to me that the compiler is likely to always copy the pointer and length because of the insane rules to observable effects (in the function call to check current log level).

I'll leave as-is for now, its not a huge penalty, but it does requiring changing the macro.